core/Enclave/key_operation.h

```diff
@@ -0,0 +1,125 @@
+                                  const ehsm_data_t *plaintext,
+                                  ehsm_data_t *cipherblob);
+
+sgx_status_t ehsm_aes_gcm_decrypt(const ehsm_data_t *aad,
+                                  const ehsm_keyblob_t *cmk,
+                                  const ehsm_data_t *cipherblob,
+                                  ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_sm4_ctr_encrypt(const ehsm_keyblob_t *cmk_blob,
+                                  const ehsm_data_t *plaintext,
+                                  ehsm_data_t *cipherblob);
+
+sgx_status_t ehsm_sm4_ctr_decrypt(const ehsm_keyblob_t *cmk_blob,
+                                  const ehsm_data_t *cipherblob,
+                                  ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_sm4_cbc_encrypt(const ehsm_keyblob_t *cmk,
+                                  const ehsm_data_t *plaintext,
+                                  ehsm_data_t *cipherblob);
+
+sgx_status_t ehsm_sm4_cbc_decrypt(const ehsm_keyblob_t *cmk,
+                                  const ehsm_data_t *cipherblob,
+                                  ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_rsa_encrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *plaintext,
+                              ehsm_data_t *ciphertext);
+
+sgx_status_t ehsm_sm2_encrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *plaintext,
+                              ehsm_data_t *ciphertext);
+
+sgx_status_t ehsm_rsa_decrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *ciphertext,
+                              ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_sm2_decrypt(const ehsm_keyblob_t *cmk,
+                              ehsm_data_t *ciphertext,
+                              ehsm_data_t *plaintext);
+
+sgx_status_t ehsm_rsa_sign(const ehsm_keyblob_t *cmk_blob,
+                           const ehsm_data_t *data,
+                           ehsm_data_t *signature);
+
+sgx_status_t ehsm_rsa_verify(const ehsm_keyblob_t *cmk,
+                             const ehsm_data_t *data,
+                             const ehsm_data_t *signature,
+                             bool *result);
+
+sgx_status_t ehsm_ec_encrypt(const ehsm_keyblob_t *cmk);
+
+sgx_status_t ehsm_ec_decrypt(const ehsm_keyblob_t *cmk);
+
+sgx_status_t ehsm_ecc_sign(const ehsm_keyblob_t *cmk,
+                           const ehsm_data_t *data,
+                           ehsm_data_t *signature,
+                           uint32_t *req_signature_len);
+
+sgx_status_t ehsm_ecc_verify(const ehsm_keyblob_t *cmk,
+                             const ehsm_data_t *data,
+                             const ehsm_data_t *signature,
+                             bool *result);
+
+sgx_status_t ehsm_sm2_sign(const ehsm_keyblob_t *cmk,
+                           const ehsm_data_t *data,
+                           const ehsm_data_t *userid,
```

Comment:
what's the userid for?

Reply:
for sm2 sign and verify

Reply:
https://github.com/guanzhi/GM-Standards/blob/master/GMT%E6%AD%A3%E5%BC%8F%E6%A0%87%E5%87%86/GMT%200009-2012%20SM2%E5%AF%86%E7%A0%81%E7%AE%97%E6%B3%95%E4%BD%BF%E7%94%A8%E8%A7%84%E8%8C%83%20.pdf
please refer to chapter 10.
user id for SM2 can be set to 1234567812345678 by default, like this:
unsigned char sm2_user_id[] = "1234567812345678";
unsigned int sm2_user_id_len = sizeof(sm2_user_id)-1;

Please remove all appid related codes in enclave.   hardcode userid in enclave as 1234567812345678